azure-pipelines/build.yml

```diff
@@ -23,29 +23,40 @@ jobs:
     displayName: 'Install Node.js'
   - script: npm ci
     displayName: 'npm ci'
-  - script: npm run build-nomaps
-    displayName: 'npm run build-nomaps'
-  - powershell: .\package.ps1
-  - task: NuGetCommand@2
+  - script: npm run build
+    displayName: 'npm run build'
+  - script: npm run webpack
+    displayName: 'npm run webpack'
+  - task: CopyFiles@2
+    displayName: 'Copy worker files to staging'
     inputs:
-      command: pack
-      packagesToPack: '$(System.DefaultWorkingDirectory)/pkg'
-      packDestination: '$(Build.ArtifactStagingDirectory)/worker'
-      ${{ if eq(parameters.IsPrerelease, true) }}:
-        buildProperties: 'prereleaseSuffix=-alpha.$(Build.BuildNumber)'
+      sourceFolder: '$(Build.SourcesDirectory)'
+      contents: |
+        dist/src/nodejsWorker.js
+        dist/src/worker-bundle.js
+        LICENSE
+        NOTICE.html
+        package.json
+        worker.config.json
+      targetFolder: '$(Build.ArtifactStagingDirectory)/worker'
+      cleanTargetFolder: true
   - script: npm prune --production
     displayName: 'npm prune --production' # so that only production dependencies are included in SBOM
   - task: ManifestGeneratorTask@0
     displayName: 'Generate SBOM for worker'
     inputs:
       BuildDropPath: '$(Build.ArtifactStagingDirectory)/worker'
-      BuildComponentPath: '$(System.DefaultWorkingDirectory)/node_modules'
+      BuildComponentPath: '$(Build.SourcesDirectory)/node_modules'
       PackageName: 'Azure Functions Node.js Worker'
-  - task: PublishBuildArtifacts@1
-    displayName: 'Publish worker artifact'
+  - task: NuGetCommand@2
+    displayName: 'NuGet pack worker'
     inputs:
-      PathtoPublish: '$(Build.ArtifactStagingDirectory)/worker'
-      ArtifactName: 'worker'
+      command: pack
+      packagesToPack: '$(Build.SourcesDirectory)/Worker.nuspec'
```

Comment:
Also wondering why we're not packing from the "staging" directory

Reply:
The basePath is the staging directory, which is where all the files are.

I could've copied the nuspec to the staging directory and then I wouldn't need to specify "packagesToPack" _and_ "basePath", but then the nuspec is listed in the manifest file which didn't seem 100% accurate considering when you unpack the nuget package you get a folder like this:
<img width="224" alt="Screen Shot 2021-12-16 at 11 55 06 AM" src="https://user-images.githubusercontent.com/11282622/146440341-00673584-0866-4457-a371-a70e0a079023.png">

It may or may not matter, but the nuspec is not in that folder so I felt like it shouldn't be in the manifest.